src/Type/MixedType.php

```diff
@@ -146,16 +146,28 @@ public function unsetOffset(Type $offsetType): Type
 
 	public function getKeysArray(): Type
 	{
+		if ($this->isArray()->no()) {
+			return new NeverType();
```

Comment:
I had the same thought and next I asked myself why we don't use that in the traits as well. Maybe because it would be ignored in a union, but actually I'm not sure. I was just wondering if we should have the same in both or not 🤔 but here it's safer to use it I guess